### PR TITLE
Fixes #37433 - Set indentation when calling subscription_manager_setup

### PR DESCRIPTION
**Pull request by stejskalleos**
Incorrect indentation in the snippet is causing yaml error in the cloud_init_default.erb

Follow up to https://github.com/theforeman/foreman/pull/10153
Detailed explanation  https://github.com/theforeman/foreman/pull/10153#issuecomment-2104528195

### CODE REVIEW

**Comment by stejskalleos** [comment]
@ShimShtein I don't know why Katello is failing, but it's not because of my changes.

**Review by ShimShtein — DISMISSED** [review]
LGTM, @ekohl any concerns?

File: test/unit/foreman/renderer/snapshots/ProvisioningTemplate/cloud-init/CloudInit_default.host4dhcp.snap.txt [hunk 1]
```diff
@@ -38,92 +38,92 @@ runcmd:
     echo "Starting the subscription-manager registration process"
   
     # Set up subscription-manager
-    # Select package manager for the OS (sets the $PKG_MANAGER* variables)
-  if [ -z "$PKG_MANAGER" ]; then
-    if [ -f /etc/os-release ] ; then
-      . /etc/os-release
+      # Select package manager for the OS (sets the $PKG_MANAGER* variables)
```
Comment (sbernhard) [inline test/unit/foreman/renderer/snapshots/ProvisioningTemplate/cloud-init/CloudInit_default.host4dhcp.snap.txt:41]: wrong indention (off by 1 space :-) )

File: test/unit/foreman/renderer/snapshots/ProvisioningTemplate/finish/Kickstart_default_finish.host4dhcp.snap.txt [hunk 2]
```diff
@@ -23,92 +23,92 @@ rpm -Uvh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
   echo "Starting the subscription-manager registration process"
 
   # Set up subscription-manager
-  # Select package manager for the OS (sets the $PKG_MANAGER* variables)
-if [ -z "$PKG_MANAGER" ]; then
-  if [ -f /etc/os-release ] ; then
-    . /etc/os-release
+    # Select package manager for the OS (sets the $PKG_MANAGER* variables)
```
Comment (sbernhard) [inline test/unit/foreman/renderer/snapshots/ProvisioningTemplate/finish/Kickstart_default_finish.host4dhcp.snap.txt]: off by 2 spaces

File: test/unit/foreman/renderer/snapshots/ProvisioningTemplate/user_data/Kickstart_default_user_data.host4dhcp.snap.txt [hunk 3]
```diff
@@ -37,92 +37,92 @@ rpm -Uvh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
...
+  subscription-manager config --rhsm.report_package_profile=1 > /dev/null 2>&1 || true
+  
+  # Configuration for EL6
+  if grep --quiet full_refresh_on_yum $RHSM_CFG; then
+    sed -i "s/full_refresh_on_yum\s*=.*$/full_refresh_on_yum = 1/g" $RHSM_CFG
+  else
+    full_refresh_config="#config for on-premise management\nfull_refresh_on_yum = 1"
+    sed -i "/baseurl/a $full_refresh_config" $RHSM_CFG
+  fi
+
```
Comment (sbernhard) [inline test/unit/foreman/renderer/snapshots/ProvisioningTemplate/user_data/Kickstart_default_user_data.host4dhcp.snap.txt:122]: one empty line to much?

File: app/views/unattended/provisioning_templates/snippet/redhat_register.erb [hunk 4]
```diff
@@ -96,7 +96,7 @@ description: |
   echo "Starting the subscription-manager registration process"
 
   # Set up subscription-manager
-  <%= snippet("subscription_manager_setup", variables: { subman_setup_scenario: 'provisioning' }).strip -%>
+  <%= indent(2) { snippet("subscription_manager_setup", variables: { subman_setup_scenario: 'provisioning' }).strip } -%>
```
Comment (sbernhard) [inline app/views/unattended/provisioning_templates/snippet/redhat_register.erb]: Can you try to set skip1 as described in https://github.com/theforeman/foreman/blob/e49650aa2f262f562efa3b024449021119699200/app/services/foreman/renderer/scope/macros/base.rb#L135 

If you have a look at https://github.com/theforeman/foreman/pull/10200/files#diff-2123697b8041a8f9d6efe3585f582d0052a40494914c861ab7b9bcb7bd0af0baR41 and the following line, the indention is off by 2 spaces

**Comment by ekohl** [comment]
@stejskalleos I think the test failures are related

**Comment by stejskalleos** [comment]
PR is not needed anymore. https://github.com/theforeman/foreman/pull/10153 has been updated and merged.

**Pull request closed without merging**